Source/Core/Core/FreelookConfig.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/FreelookConfig.h"
+#include "Core/Config/FreelookSettings.h"
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+
+namespace Freelook
+{
+Config g_Config;
+Config g_ActiveConfig;
+static bool s_has_registered_callback = false;
+
+void UpdateActiveConfig()
+{
+  g_ActiveConfig = g_Config;
+}
+
+Config::Config()
+{
+  cameraConfig.iControlType = ControlType::None;
+}
+
+void Config::Refresh()
+{
+  if (!s_has_registered_callback)
+  {
+    ::Config::AddConfigChangedCallback(
```

Comment:
This structure was copied from `VideoConfig`, I went back and forth on keeping the comment (see https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/VideoCommon/VideoConfig.cpp#L64 ) or if it'd just be confusing.  Since `UpdateActiveConfig` is still called via the video thread, I imagine the race condition could still occur but wasn't 100% sure.  Thought it'd be better to be safe than sorry.

As for why this isn't just registered in the constructor?  I'm not sure

Reply:
I can guess why `VideoConfig` does it (that is: keeping two copies; one for the UI update and another that is used by the video code; and swapping between them) - there is more than one member that might be updated during `Refresh`, and if it did that while a game was running (and unpaused) it might pick up one setting in the new state but another setting in the old state.

I suppose we can leave it at that, as long as we remember there are other places doing the same thing; for a later point if/when we change this.